plugins/CustomTCMaker.cpp

```diff
@@ -107,6 +107,15 @@ CustomTCMaker::generate_opmon_data()
   info.set_tc_failed_sent_count( m_tc_failed_sent_count.load() );  
 
   this->publish(std::move(info));
+
+  if ( m_running_flag.load() && m_latency_monitoring.load() ) { 
```

Comment:
```suggestion
  if ( m_latency_monitoring.load() && m_running_flag.load() ) { 
```
Not an important comment, I'm only writing this because I've learned something new a few weeks back and want to share :) Unnecessary micro-optimisation, but in C++ order inside if statement can matter of operations like `&&`, `||` (not true for all operators).
So in your case, whilst we're running, we will always load 2 booleans, even if we don't want monitoring. With the suggestion above, if monitoring is off, we will only load 1 boolean. Matters more task-heavy workflows so probably not here, but maybe in e.g. `TPDataProcessor` where every little counts.